pkg/infra/filestorage/filter.go

```diff
@@ -0,0 +1,244 @@
+package filestorage
+
+import (
+	"strings"
+
+	"github.com/armon/go-radix"
+	"github.com/grafana/grafana/pkg/services/accesscontrol"
+)
+
+type treeValue string
+
+const (
+	treeValueDisallowedPath   treeValue = "!"
+	treeValueDisallowedPrefix treeValue = "!*"
+	treeValueAllowedPath      treeValue = ""
+	treeValueAllowedPrefix    treeValue = "*"
+)
+
+type PathFilter interface {
+	IsAllowed(path string) bool
+	ToString() string
+
+	asSQLFilter() accesscontrol.SQLFilter
+}
+
+func NewPathFilter(allowedPrefixes []string, allowedPaths []string, disallowedPrefixes []string, disallowedPaths []string) PathFilter {
+	if allowedPrefixes != nil && allowedPaths != nil && (len(allowedPrefixes)+len(allowedPaths) == 0) {
```

Comment:
Can this be simplified to sth like:

```
if  len(allowedPrefixes) + len(allowedPaths) == 0 {
```

?

Or we need to distinguish between nil slice and initialized slice with zero length?

Reply:
Added a comment explaining the difference between `nil` and empty slices